src/Workspaces/Remote/ServiceHub/Host/RemoteExportProvider.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.IO;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Extensions;
+using Microsoft.CodeAnalysis.ExternalAccess.AspNetCore.Internal.EmbeddedLanguages;
+using Microsoft.CodeAnalysis.ExternalAccess.Razor;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.Composition;
+
+namespace Microsoft.CodeAnalysis.Remote;
+
+internal static class RemoteExportProvider
+{
+    internal static readonly ImmutableArray<Assembly> RemoteHostAssemblies =
+        MefHostServices.DefaultAssemblies
+            .Add(typeof(AspNetCoreEmbeddedLanguageClassifier).Assembly)
+            .Add(typeof(BrokeredServiceBase).Assembly)
+            .Add(typeof(IRazorLanguageServerTarget).Assembly)
+            .Add(typeof(RemoteWorkspacesResources).Assembly)
+            .Add(typeof(IExtensionWorkspaceMessageHandler<,>).Assembly);
+
+    private static ExportProvider? s_instance;
+    internal static ExportProvider ExportProvider
+    {
+        get
+        {
+            Contract.ThrowIfNull(s_instance, "Default export provider not initialized. Call InitializeAsync first.");
+            return s_instance;
+        }
+    }
+
+    public static async Task InitializeAsync(string localSettingsDirectory, CancellationToken cancellationToken)
+    {
+        var args = new ExportProviderBuilder.ExportProviderCreationArguments(
+            AssemblyPaths: RemoteHostAssemblies.SelectAsArray(static a => a.Location),
+            Resolver: new Resolver(SimpleAssemblyLoader.Instance),
+            CacheDirectory: Path.Combine(localSettingsDirectory, "Roslyn", "RemoteHost", "Cache"),
+            CatalogPrefix: "RoslynRemoteHost",
+            ExpectedErrorParts: ["PythiaSignatureHelpProvider", "VSTypeScriptAnalyzerService", "RazorTestLanguageServerFactory", "CodeFixService"],
+            PerformCleanup: true,
+            LogError: static _ => { },
+            LogTrace: static _ => { });
```

Comment:
why no logging?  just no way to support it?  it seems like this call could be passed an RPC callback to send this info.

Reply:
I didn't see an obvious way to pipe things through. Logging this early in the remote host initialization process isn't trivial.

Reply:
Can we at least write to a string builder and send the string back from InitializeAsync? And propagate that somewhere if the error string is non-empty?

Reply:
Went ahead and pushed back any error messages occurred during the remote initialize call.